pkg/oc/bootstrap/docker/up.go

```diff
@@ -406,22 +404,27 @@ func (c *CommonStartConfig) Complete(f *osclientcmd.Factory, cmd *cobra.Command,
 		taskPrinter.Success()
 	}
 
-	// TODO HOLY SIDE-EFFECTS this is setting global port values as part of a check!
 	// Ensure that ports used by OpenShift are available on the host machine
 	taskPrinter.StartTask("Checking for available ports")
-	if err := c.CheckAvailablePorts(out); err != nil {
+	useAlternativeDNSPort, err := c.CheckAvailablePorts(out)
```

Comment:
Let's just always assume we start with 8053 and never have an alternate.  Then this `CheckAvailablePorts` moves to the `Check` function.